### PR TITLE
Data loading reafactor and improved fetch parallelism

### PR DESCRIPTION
**Pull request by rgembalik**
### Rationale

Because we are often using the plugin with large datasets (over 20 000), I thought it mandatory to have a proper download job management and download limits. 

### Please, help with testing
It goes without saying, that the PR touches on crucial parts of the plugin. I'd feel better if someone else gave it a go and test it in different projects. I tested most of the basic functions (data loading, data loading with cache filled, data removal and updates), but I am sure I missed some things.

### Functional improvements

- `singleFetchLimit` - an option, which forces data download in chunks. default: `1000`, max `5000`
- `maxConcurrentDataDownloads` - an option which limits the number of concurrent downloads. Default: `10`

Please review those default values - I think we might want to reduce them further

### Engine improvements

- The data now is downloaded by workers, using a simple download queue. The queue is populated first when downloading one page from each Content Type, and then (once we know the number of objects to be downloaded), the rest of the jobs is being filled into the queue

### Refactoring

This was a good opportunity to start splitting the code by responsibility as I had to rewrite the data loading anyway, so:
- Data loading methods (content types, content objects, content object removals) are extracted into `src/data-loader.js`. Currently, they are gatsby-aware to be able to check the number of nodes in each content type etc. However, these are not registering new nodes, nor removing them - I kept that intentionally outside. 
- Some utility functions were extracted into `src/utils.js` - mostly for the sake of re-usage in `data-loader.js`
- Workers mini-library was created in `src/workers.js`

The refactoring process is not finished in my opinion, but I didn't want to move everything at once, so I stopped at the level I felt manageable in single PR.

### CODE REVIEW

File: src/data-loader.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            page: 1,
+            ctd
+        }
+    });
+
+    const dataLoadCount = {}
+
+    await workers(downloadJobs, maxConcurrentDataDownloads, async ({baseUrl, objectLimit, page, ctd, totalPages, limitPerPage}) => {
+        const url = `${baseUrl}&page=${page}`;
+        const humanizedPageNumber = page === 1 ? 'frist page' : `${page}/${totalPages}`
```
Comment (CiotkaCierpienia) [inline src/data-loader.js:103]: Typo in first page

**Comment by CiotkaCierpienia** [comment]
Can you please add information about new properties in the Readme?

**Comment by rgembalik** [comment]
> Can you please add information about new properties in the Readme?

Yep, forgot about that. I added the description in the readme. Also, to introduce some kind of limitations within the plugin, I introduced proper min/max values for the new options

**Comment by CiotkaCierpienia** [comment]
We've added default value for apiUrl in master, can you make sure that the default value is also used in refactored code?

**Pull request merged**